BierBuddy.UILib/ProfilePageRenderer.cs

```diff
@@ -101,12 +101,34 @@ private Grid GetNameAndAgeLabels()
             grid.ColumnDefinitions.Add(new ColumnDefinition { Width = new GridLength(1, GridUnitType.Star) });
             grid.RowDefinitions.Add(new RowDefinition { Height = new GridLength(1, GridUnitType.Star) });
 
-            ProfileContentBorder name = new ProfileContentBorder(_Visitor.Name, UIUtils.BabyPoeder, GeneralFontSize);
-            name.ProfileContentLabel.Foreground = Brushes.Black;
-            name.Margin = new Thickness(0, 0, 20, 0);
-            name.ProfileContentLabel.HorizontalAlignment = HorizontalAlignment.Left;
-            name.ProfileContentLabel.Padding = new Thickness(20, 10, 0, 10);
-            name.VerticalAlignment = VerticalAlignment.Center;
+            ProfileContentBorder name;
+            if (_ReadOnly)
+            {
+                name = new ProfileContentBorder(_Visitor.Name, UIUtils.BabyPoeder, GeneralFontSize);
+                name.ProfileContentLabel.Foreground = Brushes.Black;
+                name.Margin = new Thickness(0, 0, 20, 0);
+                name.ProfileContentLabel.HorizontalAlignment = HorizontalAlignment.Left;
+                name.ProfileContentLabel.Padding = new Thickness(20, 10, 0, 10);
+                name.VerticalAlignment = VerticalAlignment.Center;
+            } else
+            {
+                name = new ProfileContentBorder(UIUtils.BabyPoeder);
+                name.Margin = new Thickness(0, 0, 20, 0);
+                name.VerticalAlignment = VerticalAlignment.Center;
+                TextBox textBox = GetTransparentTextBox();
+                textBox.Text = _Visitor.Name;
+                textBox.Margin = new Thickness(20, 10, 0, 10);
+                textBox.TextChanged += (s, e) =>
+                {
+                    _Visitor.Name = textBox.Text;
```

Comment:
het aanpassen van een visitor name is eigenlijk(in de strengste opvatting) businesslogic, kan je deze niet beter in de profilepage afhandellen ipv de renderer. dan kan je ook een input check dr overheen gooien.

Reply:
top, dan allen de verandering van de variablele in de businesslogic doen